### ISSUE TITLE
Continuous deployment by branch for datagov-deploy in BSP

### ISSUE THREAD

**Issue by adborden** [GSA/data.gov#2705]
## User Story

In order to deploy develop, release, and production branches with datagov-deploy, the Data.gov needs to support branch based builds in Jenkins for any platform deploys.

## Acceptance Criteria

- [ ] WHEN I push a release branch to GitHub for datagov-deploy
  AND it passes our continuous integration tests
  THEN Jenkins will use the release branch of datagov-deploy for the deploy

## Background
I had this partially implemented pretty naively in the Jenksinfile. Then I realized that because the application deploys use the datagov-deploy:latest image tag, they will use the datagov-deploy code from the most recent build, which could be a release branch when it should be using master. This story is about properly implementing that.

The CircleCI job (and bin/jenkins_build) will trigger a platform build with a branch_name parameter. The rest of this story is about making sure build artifacts (the docker container images) are tagged by branch and that the Jenkinsfile in datagov-deploy (and maybe the applications) are invoking the correct tag.

## Security Considerations ([required](https://nvd.nist.gov/800-53/Rev4/control/CM-4))
[comment]: # "Our SSP says 'The Data.gov team ensures security implications are considered as part of the agile requirements refinement process by including a section in the issue template used as a basis for new work.' so please don't remove this section without care."
[Any security concerns that might be implicated in the change. "None" is OK, just be explicit here!]

## Sketch
- [ ] Move the docker commands to a script to avoid too much logic in groovy or the Jenkinsfile.
- [ ] Build the docker image and tag it based on the branch. For master branch, use latest.
- [ ] Install the script to the Jenkins instance so it is available to all build jobs.
- [ ] Update Jenkinsfile in datagov-deploy and/or application repos if necessary.

**Comment by adborden** [comment]
Parking this work on the [feature/jenkins-branch branch](https://github.com/GSA/datagov-deploy/tree/feature/jenkins-branch).

I started working on https://github.com/GSA/datagov-deploy/issues/2708 to drop docker, somewhat rolling this story into it.

After the conversation in sprint planning, I realized maybe we don't need this story at all, we can assume to build from only 1 branch (`master`/`main`) since deploys will still deploy to staging, then production. A deploy is not necessary from the release/* branches.
